doc/fwu/api/api.rst

```diff
@@ -557,13 +557,40 @@ The following functions are used to prepare a new firmware image in the componen
 
    To abandon an update that has been started, call `psa_fwu_cancel()`, and then `psa_fwu_clean()`.
 
+.. macro:: PSA_FWU_LOG2_WRITE_ALIGN
+   :definition: /* implementation-defined value */
+
+   .. summary::
+      Base-2 logarithm of the required alignment of firmware image data blocks when calling `psa_fwu_write()`.
+
+   This value specifies the minimum alignment of a data block within a firmware image, when written using `psa_fwu_write()`. The value is the base-2 log of the alignment size. `PSA_FWU_LOG2_WRITE_ALIGN` is used to constrain the values of ``image_offset`` that are supported, and the handling of a data block of unaligned size, as follows:
+
+   *  Let :code:`WRITE_ALIGN_MASK = (1<<PSA_FWU_LOG2_WRITE_ALIGN) - 1`
+   *  If :code:`(image_offset & WRITE_ALIGN_MASK) != 0`, then the implementation returns :code:`PSA_ERROR_INVALID_ARGUMENT`.
+   *  If :code:`(block_size & WRITE_ALIGN_MASK) != 0`, then the implementation will pad the data with :scterm:`implementation defined` values up to the next aligned size, before writing the data to the firmware image.
+   *  This value does **not** constrain the alignment of the data buffer, ``block``.
+
+   The specific value of `PSA_FWU_LOG2_WRITE_ALIGN` is an :scterm:`implementation defined`, non-negative integer. If an implementation has no alignment requirement, then it defines `PSA_FWU_LOG2_WRITE_ALIGN` to be ``0``.
+
+   .. admonition:: Implementation note
+
+      It is recommended that `PSA_FWU_LOG2_WRITE_ALIGN` is not greater than ``17``, which corresponds to a block size of 128 KB.
```

Comment:
Does it make sense to explain why?

Reply:
Should we explicitly mention the file delivery protocol that AWS uses for MQTT connections? - or can we have some more general wording about 'ensuring compatibility with existing block-based file transfer protocols used for IoT devices'?